app/services/cloud_topology_service.rb

```diff
@@ -9,6 +9,8 @@ class CloudTopologyService < TopologyService
     :cloud_tenants      => [:tags, :vms => :tags],
   ]
 
+  @kinds = %i(CloudManager AvailabilityZone CloudTenant Vm Tag)
```

Comment:
```
def kinds
  %i(CloudManager AvailabilityZone CloudTenant Vm Tag
end
```

.. then you don't need a magic `self.class.instance_variable_get`, and you can just use `kinds` :)